Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -671,6 +671,11 @@ ElementUpdate TreeResolver::createAnimatedElementUpdate(ResolvedStyle&& resolved
         // such that when CSS Transitions are updated the CSS Animations data is the same as during the previous style change event.
         if ((oldStyle && oldStyle->hasAnimations()) || resolvedStyle.style->hasAnimations())
             styleable.updateCSSAnimations(oldStyle, *resolvedStyle.style, resolutionContext, newStyleOriginatedAnimations, isInDisplayNoneTree);
+
+        if (((oldStyle && oldStyle->scrollTimelines().size()) || resolvedStyle.style->scrollTimelines().size())
+            || ((oldStyle && oldStyle->scrollTimelineNames().size()) || resolvedStyle.style->scrollTimelineNames().size())) {
```

Comment:
Should we be calling this even when the scroll timeline names didn't change?

Reply:
I left a `FIXME` in `Styleable::updateCSSScrollDrivenTimelines()` about that. I'm looking into address this now.

Reply:
This is addressed with the latest commit.